### ISSUE TITLE
--format flag no longer available as of 0.43.0

### ISSUE THREAD

**Issue by nmccann** [realm/SwiftLint#3571]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Previous versions of the tool were able to autocorrect the format of linted code when passing the `--format` flag, for example, `swiftlint autocorrect --format`. This no longer seems possible as of 0.43.0, but worked in 0.42.0, likely related to the move from `autocorrect` to `--fix`. I checked the README, as well as the release notes, and didn't see any information indicating that this flag was intentionally removed, or relocated. If someone can point me to such information (if it exists), that would be great.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
swiftlint autocorrect --format
The `swiftlint autocorrect` command is no longer available.
Please use `swiftlint --fix` instead.
Error: Unknown option '--format'
Usage: swiftlint lint <options>
  See 'swiftlint lint --help' for more information.
```

```bash
swiftlint --fix --format
Error: Unknown option '--format'
Usage: swiftlint lint <options>
  See 'swiftlint lint --help' for more information.
```

I also tried a few different variations with `--format` or just `format`, some of which didn't throw errors, but also didn't correct formatting issues.

### Environment

* SwiftLint 0.43.0
* CocoaPods
* Xcode 12.4

**Comment by AnarionII** [comment]
I'm experiencing the same issue. 

Here is my use case: I'm using swiftlint to format the code on build with this Build Phases / Run Script : 
```
${PODS_ROOT}/SwiftLint/swiftlint --fix --config ../.swiftlint.yml --path ./ --format
${PODS_ROOT}/SwiftLint/swiftlint lint --config ../.swiftlint.yml --path ./
```

The lack of this flag in 0.43.0 force me to stay in 0.42.0 Please could you add it back?

**Comment by jpsim** [comment]
I must have missed this when migrating from Commandant to swift-argument-paser in https://github.com/realm/SwiftLint/pull/3465. It should be easy to add back in.

There were release candidates that included this migration starting in [January](https://github.com/realm/SwiftLint/releases/tag/0.43.0-rc.1) and others in February, leading up to the official release in [March](https://github.com/realm/SwiftLint/releases/tag/0.43.0).

The changelog specifically called out the possibility of breaking changes:

> Previous commands should continue to work temporarily to help with the transition. Please let us know if there's a command that no longer works and we'll attempt to add a bridge to help with its transition.

And I asked for testers on Twitter in [February](https://twitter.com/simjp/status/1362164514851074050):

> I encourage you to try this version and share how it goes since it includes some potentially breaking changes to the CLI interface.

No one reported this missing flag.

Is there anything that would help you (or users in general) actually install and evaluate release candidates in the future?

**Comment by AnarionII** [comment]
Thank you for the answer JP. My bad, next time I will test RCs to report bugs earlier 😇.

BTW: you're doing an amazing work 💪🏻, SwiftLint is an amazing tool.

**Comment by jpsim** [comment]
I've re-added the flag in https://github.com/realm/SwiftLint/pull/3573.

@nmccann @AnarionII could you please try that branch and report back if the feature is working as expected for you?

Note that the `autocorrect` command was deprecated in favour of a `--fix` flag on the `lint` and `analyze` commands, but the following should all work:

* `swiftlint autocorrect --format`
* `swiftlint --fix --format`
* `swiftlint lint --fix --format`

**Comment by AnarionII** [comment]
> I've re-added the flag in #3573.
> 
> @nmccann @AnarionII could you please try that branch and report back if the feature is working as expected for you?

I'm on it.

> 
> Note that the `autocorrect` command was deprecated in favour of a `--fix` flag on the `lint` and `analyze` commands, but the following should all work:
> 
> * `swiftlint autocorrect --format`
> * `swiftlint --fix --format`
> * `swiftlint lint --fix --format`

Thanks for the tip

**Comment by nmccann** [comment]
@jpsim thank you for the quick response and fix - I can confirm that your branch resolves the issue (also tried the alternative commands, different indentation widths, and with tabs - all worked).

I apologize for not testing prior to the release of 0.43, I hope my bug report didn't come off as sounding entitled - the release notes for 0.43 did make it quite clear that issues might arise, and I was content with reverting to 0.42 as there was no critical reason to update. I compared the diff between 0.43 and 0.42 to see if I could identify the issue, but the large number of changes going from Commandant to swift-argument-parser made it difficult to grok where things may have gone awry. I appreciate the continued work on this project by yourself and others.

@AnarionII - have you found autocorrecting on build to be a good addition to your workflow? I considered doing this in the past, but found it problematic as corrected files lose their undo history (though files that aren't corrected keep it). Perhaps you found a workaround for this, or perhaps the impact isn't as severe as I thought it would be.

**Comment by jpsim** [comment]
> I hope my bug report didn't come off as sounding entitled

Not at all, I'm mostly just curious how to increase the number of release candidate testers. I haven't been able to come up with a good incentive, and unsurprisingly "help catch bugs before they ship" isn't terribly compelling.

**Comment by jpsim** [comment]
> have you found autocorrecting on build to be a good addition to your workflow? I considered doing this in the past, but found it problematic as corrected files lose their undo history (though files that aren't corrected keep it). Perhaps you found a workaround for this, or perhaps the impact isn't as severe as I thought it would be.

I'd advise against this, because not only does Xcode lose undo history for a file that's modified outside of the editor, but if there's any delay between hitting build in Xcode and when the corrections are actually calculated and applied, if you edit the file in the meantime, SwiftLint might apply its corrections to the wrong place, leading to garbled results.

**Comment by jpsim** [comment]
Please update to https://github.com/realm/SwiftLint/releases/tag/0.43.1 which re-adds this flag.

**Comment by AnarionII** [comment]
Thank you so much JP for this quick fix, it works like a charm.

> have you found autocorrecting on build to be a good addition to your workflow? I considered doing this in the past, but found it problematic as corrected files lose their undo history (though files that aren't corrected keep it). Perhaps you found a workaround for this, or perhaps the impact isn't as severe as I thought it would be.

> I'd advise against this, because not only does Xcode lose undo history for a file that's modified outside of the editor, but if there's any delay between hitting build in Xcode and when the corrections are actually calculated and applied, if you edit the file in the meantime, SwiftLint might apply its corrections to the wrong place, leading to garbled results.

I mainly use linters for one reason: to avoid most of the merge conflicts by standardizing as much a possible the source code between branches/developers. So 50% of the work is done by the formatter and 50% by the linter it self. So even with the issues you both described, according to me it's still worth it to format during build phase.
